transport/tchannel/handler.go

```diff
@@ -90,31 +92,39 @@ func (h handler) Handle(ctx ncontext.Context, call *tchannel.InboundCall) {
 }
 
 func (h handler) handle(ctx context.Context, call inboundCall) {
-	start := time.Now()
-	err := h.callHandler(ctx, call, start)
-	if err == nil {
-		return
+	// you MUST close the responseWriter no matter what
+	responseWriter := newResponseWriter(call.Response(), call.Format())
+
+	handlerErr := h.callHandler(ctx, call, responseWriter)
+	if handlerErr != nil {
+		// we have an error, so we're going to propagate it as a yarpc error,
+		// regardless of whether or not it is a system error.
```

Comment:
This doesn't sound right. Looks like callHandler returns a TChannel-specific error which should be propagated as an error frame (system error) rather than a call res.